### PR TITLE
Binary tree problem, height of Binary tree solution added

### PR DESCRIPTION
**Pull request by kondekarshubham123**
### Describe your change:

Height of binary tree using dfs, solution added.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/binary_tree/height_of_binary_tree.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+def height_of_tree(root: TreeNode | None, depth=0) -> int:
```
[the remark this answers is not in the document]
Comment (kondekarshubham123) [reply data_structures/binary_tree/height_of_binary_tree.py:16]: type hint for depth param added

File: data_structures/binary_tree/height_of_binary_tree.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Given the root of a binary tree, return its maximum depth.
+# A binary tree's maximum depth is the number of nodes along the longest
+# path from the root node down to the farthest leaf node.
+
+
+from typing import Optional
+
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
```
[the remark this answers is not in the document]
Comment (kondekarshubham123) [reply data_structures/binary_tree/height_of_binary_tree.py:10]: None type added

File: data_structures/binary_tree/height_of_binary_tree.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+def height_of_tree(root: Optional[TreeNode], depth=0) -> int:
```
[the remark this answers is not in the document]
Comment (kondekarshubham123) [reply data_structures/binary_tree/height_of_binary_tree.py]: updated with depth type hint param

File: data_structures/binary_tree/height_of_binary_tree.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+class TreeNode:
+    def __init__(
+        self, val: int = 0, left: TreeNode | None = None, right: TreeNode | None = None
+    ):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+def height_of_tree(root: TreeNode | None, depth=0) -> int:
```
[the remark this answers is not in the document]
Comment (kondekarshubham123) [reply data_structures/binary_tree/height_of_binary_tree.py:17]: type hint for parameter depth added

File: data_structures/binary_tree/height_of_binary_tree.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+     /  \
+    15   7  ->  3
+"""
+tree_1 = TreeNode(3)
+tree_1.left = TreeNode(9)
+tree_1.right = TreeNode(20)
+tree_1.right.left = TreeNode(15)
+tree_1.right.right = TreeNode(7)
+
+print(height_of_tree(tree_1))  # Output: 3
```
Comment (CenTdemeern1) [inline data_structures/binary_tree/height_of_binary_tree.py:49]: Could you please move this example into the `if __name__ == "__main__":` block or into a doctest so it doesn't run on import?

**Review by CenTdemeern1 — CHANGES_REQUESTED** [review]

File: data_structures/binary_tree/height_of_binary_tree.py (hunk 5 above)
Comment (kondekarshubham123) [reply data_structures/binary_tree/height_of_binary_tree.py]: example moved into if __name__ == "__main__": block

**Review by CenTdemeern1 — APPROVED** [review]

File: data_structures/binary_tree/height_of_binary_tree.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# Given the root of a binary tree, return its maximum depth.
+# A binary tree's maximum depth is the number of nodes along the longest
+# path from the root node down to the farthest leaf node.
+
+from __future__ import annotations
+
+
+class TreeNode:
```
Comment (cclauss) [inline data_structures/binary_tree/height_of_binary_tree.py:8]: dataclass?

**Comment by kondekarshubham123** [comment]
@cclauss, seems this problem is already covered here

https://github.com/TheAlgorithms/Python/blob/3448ae5cec868d4a03349cb952765e9abff41243/data_structures/binary_tree/binary_tree_traversals.py#L61

**Comment by kondekarshubham123** [comment]
I saw this in sometime before only, I think we can discard this PR
Sorry for that

Will try to add the questions not added in Repo.

**Comment by Ghostcell3** [comment]
sipa:sipa

**Pull request closed without merging**